sympy/series/limits.py

```diff
@@ -206,13 +206,13 @@ def doit(self, **hints):
             return Order(limit(e.expr, z, z0), *e.args[1:])
 
         try:
-            r = gruntz(e, z, z0, dir)
-            if r is S.NaN:
-                raise PoleError()
-        except (PoleError, ValueError):
             r = heuristics(e, z, z0, dir)
             if r is None:
                 return self
+        except (PoleError, ValueError):
+            r = gruntz(e, z, z0, dir)
+            if r is S.NaN:
+                raise PoleError()
```

Comment:
For what is this PoleError raised?

Reply:
@normalhuman this PoleError was also raised in the original code